gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/remote/DriverRemoteTraversalSideEffects.java

```diff
@@ -39,12 +40,14 @@
 public class DriverRemoteTraversalSideEffects extends AbstractRemoteTraversalSideEffects {
 
     private final Client client;
-    private Set<String> keys = null;
+    private Set<String> keys = Collections.emptySet();
     private final UUID serverSideEffect;
     private final Host host;
 
     private final Map<String, Object> sideEffects = new HashMap<>();
 
+    private boolean closed = false;
```

Comment:
I suppose it is ok that the close status isn't really thread safe. I guess it would be weird for someone to make multi-threaded calls over the side-effect object. That seems unusual. Just thought i'd point that out.